modules/core/shared/src/main/scala/coursier/core/Version.scala

```diff
@@ -188,7 +188,7 @@ object Version {
 
       if (isNumeric(item)) {
         val nextNonDotZero = _tokens.dropWhile{case (Tokenizer.Dot, n: Numeric) => n.isEmpty; case _ => false }
-        if (nextNonDotZero.forall(t => t._1 == Tokenizer.Hyphen || ((t._1 == Tokenizer.Dot || t._1 == Tokenizer.None) && !isNumeric(t._2)))) { // Dot && isNumeric(t._2)
+        if (nextNonDotZero.forall(t => !isMinMax(t._2) && (t._1 == Tokenizer.Hyphen || ((t._1 == Tokenizer.Dot || t._1 == Tokenizer.None) && !isNumeric(t._2))))) { // Dot && isNumeric(t._2)
```

Comment:
Are the changes in this file really necessary? Seems everything still works fine without those.

Reply:
Yes, they are necessary if you have a zero in your version at some place. `1.2.0-+` should mean `1.2.0-SOMETHING`, but with "normal" versions, `1.2.0` is equivalent to `1.2` and thus the zero is redacted. If you add a `-max` to that number you have `1.2-max` which has a totally different meaning than desired.

Reply:
Add a test case for it then?

Reply:
Sure thing.